src/Raven.Server/Documents/Indexes/IndexStore.cs

```diff
@@ -353,87 +352,65 @@ public IEnumerable<Index> GetIndexes()
-                        {
-                            // If it's a fairly established query then we need to determine whether there is any activity currently
-                            // If there is activity and this has not been queried against 'recently' it needs idling
-
-                            var nextItem = autoIndexesSortedByLastQueryTime[i + 1];
-                            if (nextItem.LastQueryingTime - item.LastQueryingTime > timeToWaitBeforeMarkingAutoIndexAsIdle.AsTimeSpan)
-                                item.Index.SetPriority(IndexingPriority.Idle);
-                        }
-
-                        continue;
-                    case IndexingPriority.Idle:
-                        HandleIdleAutoIndex(item.Index, age, lastQuery, timeToWaitBeforeMarkingAutoIndexAsAbandoned);
-                        break;
-                }
-            }
-        }
-
-        private static void HandleActiveAutoIndex(Index index, TimeSpan age, TimeSpan lastQuery, TimeSetting timeToWaitBeforeMarkingAutoIndexAsIdle)
-        {
-            var timeToWaitForIdle = timeToWaitBeforeMarkingAutoIndexAsIdle.AsTimeSpan.TotalMinutes;
-
-            if (age.TotalMinutes < timeToWaitForIdle * 2.5 && lastQuery.TotalMinutes < 1.5 * timeToWaitForIdle)
-                return;
-
-            if (age.TotalMinutes < timeToWaitForIdle * 6 && lastQuery.TotalMinutes < 2.5 * timeToWaitForIdle)
-                return;
+                    TimeSpan differenceBetweenNewestAndCurrentQueryingTime;
+                    if (i < indexesSortedByLastQueryTime.Count - 1)
+                    {
+                        var lastItem = indexesSortedByLastQueryTime[indexesSortedByLastQueryTime.Count - 1];
+                        differenceBetweenNewestAndCurrentQueryingTime = lastItem.LastQueryingTime - item.LastQueryingTime;
+                    }
+                    else
+                        differenceBetweenNewestAndCurrentQueryingTime = TimeSpan.Zero;
+
+                    if (differenceBetweenNewestAndCurrentQueryingTime > timeToWaitBeforeMarkingAutoIndexAsIdle.AsTimeSpan)
+                    {
+                        if (lastQuery > timeToWaitBeforeMarkingAutoIndexAsIdle.AsTimeSpan)
+                            item.Index.SetPriority(IndexingPriority.Idle);
+                    }
 
-            index.SetPriority(IndexingPriority.Idle);
-        }
+                    continue;
+                }
 
-        private void HandleIdleAutoIndex(Index index, TimeSpan age, TimeSpan lastQuery, TimeSetting timeToWaitBeforeMarkingAutoIndexAsAbandoned)
-        {
-            // relatively young index, haven't been queried for a while already
-            // can be safely removed, probably
-            if (age.TotalMinutes < 90 && lastQuery.TotalMinutes > 30)
-            {
-                DeleteIndex(index.IndexId);
-                return;
+                if (item.Priority.HasFlag(IndexingPriority.Idle))
```

Comment:
We have to have detailed logging here. 
Index state moved to idle, index was deleted for inactivity, etc.